File: day2/2.py

def invalid(i):
    num = str(i)

    for i in range(1, len(num)):
        if (len(num) % i): continue
        flag = True
        j = 0
        while (j < len(num)):
            if (num[:i] != num[j:j+i]):
                flag = False
            j += i
        if flag:
            return True
    return False

cnt = 0
with open('day2/input.txt', 'r') as file:
    for line in file:
        ranges = line.strip().split(",")
        for ran in ranges:
            st, end = map(int, ran.split("-"))
            for i in range(st, end + 1):
                if (invalid(i)):
                    print(i)
                    cnt += i

print(cnt)

File: day5/2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    vector<pair<long long, long long>> ranges;
    long long cnt = 0;

    string filename = "input.txt";
    ifstream inputFile(filename);

    string line;
    while (getline(inputFile, line)) {
        if (line == "") break;
        vector<long long> v;
        string num;
        istringstream numstream(line);
        while (getline(numstream, num, '-')) {
            v.push_back(stoll(num));
        }
        ranges.push_back({v[0], v[1]});
    }

    sort(ranges.begin(), ranges.end());
    cnt += ranges[0].second - ranges[0].first + 1;
    long long maxe = ranges[0].second;
    for (int i = 1; i < ranges.size(); i++) {
        if (maxe >= ranges[i].second) {
            continue;
        } else if (maxe >= ranges[i].first) {
            cnt += ranges[i].second - maxe;
        } else {
            cnt += ranges[i].second - ranges[i].first + 1;
        }
        maxe = max(maxe, ranges[i].second);
    }
    inputFile.close();
    cout << cnt << endl;
}